packages/savings-vault/src/lib/savings-vault-router.ts

```diff
@@ -232,17 +232,23 @@ export class SavingsVaultRouter implements Router {
 		amount: BigNumberish,
 		outputTokenAddress?: Address,
 		options?: Partial<{
-			maxLoss: BigNumber;
+			maxLoss: number;
 		}>
 	): Promise<TransactionResponse> {
 		const owner = await savingsVault.account.address();
 
 		if (options?.maxLoss) {
-			return savingsVault.redeemWithMaxLoss(
+			if (options.maxLoss < 0 || options.maxLoss > 10000)
+				throw new RangeError('Parameter maxLoss must be between 0 and 10000.');
+
+			const redeemed = await savingsVault.contract.previewRedeem(amount);
+			const maxOutputAmount = redeemed.mul(10000 + options.maxLoss).div(10000);
```

Comment:
It's the opposite of this, so `minOuputAmount = redeemed.mul(options.maxLoss).div(10_000)`.

Reply:
oops, yes, you're right

Reply:
I've initially done just like that, but than was thinking if division by zero case